fusionauth/resource_fusionauth_application.go

```diff
@@ -686,10 +718,26 @@ func newOAuthConfiguration() *schema.Resource {
 				Default:     false,
 				Description: "When enabled the user will be required to be registered, or complete registration before redirecting to the configured callback in the authorization code grant or the implicit grant. This configuration does not currently apply to any other grant.",
 			},
-			"provided_scope_policy": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Elem:     newOAuthConfigurationScopePolicy(),
+			"scope_handling_policy": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     fusionauth.OAuthScopeHandlingPolicy_Strict.String(),
```

Comment:
While this matches the API defaults, should we make it default to Compatibility to prevent breaking existing TF configs? 

It may be possible to add some special handling so that it only sets it to Strict on create, and not overwrite existing applications.

Reply:
I think that it's good to keep the same default between the API and TF. It wasn't immediately obvious how to customize the default, but I would assume it's possible. It sounds like the `Default` here is static and cannot be customized.

[CustomizeDiff](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/customizing-differences) could be useful, but that may only be used to decide whether changes need to be applied.

The issue is still present in older versions of the provider. I believe that's because the provider uses `PUT` instead of `PATCH` methods on the client calls. With no value provided, it will use the default of `Strict` in the API.

The data migration for `1.50.0` includes setting existing applications to `Compatibility` mode. Would it be unreasonable to expect provider users to update their TF state after an upgrade? If the new fields are available on the provider resource, I would expect that updating the state would pull in the migrated values from the application and then continue setting them when the configuration is applied in the future.